packages/seacas/libraries/ioss/src/Ioss_DatabaseIO.h

```diff
@@ -545,6 +545,10 @@ namespace Ioss {
       }
     }
 
+    virtual std::vector<size_t> get_all_block_connectivity(const std::string &field_name, void *data, size_t data_size) const;
+    virtual std::vector<size_t> get_all_block_field_data(const std::string &field_name,
+                                                         void *data, size_t data_size) const;
```

Comment:
The base implementation does look easy to extend to general grouping entity types .. I don't know how easy this would be for the parallel decomp case since that relies heavily on `Ioss::BlockDecompositionData` which is element block based. I don't see an equivalent decomposition for the other grouping entity types ... if we could come up with a general `Ioss::GroupingEntityDecompositionData` type that could handle other grouping entity types such as sidesets, etc , then we would have a definite shot at extending this